doc/user/content/integrations/terraform.md

```diff
@@ -0,0 +1,223 @@
+---
+title: "Terraform"
+description: "Create and manage Materialize objects with the Terraform"
+menu:
+  main:
+    parent: "integrations"
+    weight: 11
+    name: "Terraform"
+---
+
+[Terraform](https://www.terraform.io/) is an infrastructure-as-code tool that allows you to manage your
+resources in a declarative configuration language. Materialize maintains a
+[Terraform provider](https://registry.terraform.io/providers/MaterializeInc/materialize/latest/docs) to help you configure Materialize resources like
+connections, sources, and database objects. The Materialize provider is a
+Terraform plugin that allows Terraform to interact with the Materialize API.
+
+Materialize also maintains several modules that make it easier to manage your
+other cloud resources that Materialize depends on. Modules allow you to bypass
+manually configuring cloud resources and are an efficient way of deploying
+infrastructure with a single `terraform apply` command.
+
+## Use cases
+
+Materialize objects can be created through standard SQL queries, but managing
+more than a few components can be difficult to maintain.  Using Terraform can be
+helpful when you manage other cloud resources in collaboration with Materialize.
+The Terraform modules below provide the cloud infrastructure foundation
+Materialize needs to communicate with components outside of Materialize itself.
+The Materialize provider allows users to manage Materialize resources in the
+same programmatic way.
+
+You can use the modules to establish the underlying cloud
+resources and then use the Materialize provider to build Materialize-specific
+objects. A few use cases are captured in the sections below:
+
+### MSK or self-managed Kafka and PrivateLink
+
+To get data into Materialize, you need a connection to allow your data source to
+communicate with Materialize. One option to connect securely to Materialize is
+AWS PrivateLink.
+
+The [AWS MSK PrivateLink](https://github.com/MaterializeInc/terraform-aws-msk-privatelink) and [AWS Kafka PrivateLink](https://github.com/MaterializeInc/terraform-aws-kafka-privatelink) modules allow you to manage
+your connection to Materialize in a single configuration. The module builds
+target groups for brokers, a network load balancer, a TCP listener, and a VPC. These AWS resources are necessary components for creating a PrivateLink connection.
```

Comment:
One small remark, we don't create a new VPC, we use the existing VPC that the MSK/Kafka clusters are in. The modules create a **VPC endpoint service** which is actually the AWS Privatelink.